experimental/dds/tree2/src/feature-libraries/editable-tree/editableField.ts

```diff
@@ -47,13 +48,34 @@ import {
 import { makeTree } from "./editableTree";
 import { ProxyTarget } from "./ProxyTarget";
 
+/* eslint-disable no-param-reassign */
+/* eslint-disable no-bitwise */
+
+/**
+ * Coerces the given 'index' to an Int32.  This includes the following behaviors:
+ *
+ * - +/-Infinity is preserved
+ * - \{ undefined, null, false, NaN \} map to 0
+ * - \{ true \} maps to 1
+ * - Strings containing numeric digits are parsed (e.g. "-0.1e2" -\> -10)
+ * - Non-parsable strings are coerced to 0
+ *
+ * If the resulting integer is negative, counts backwards from the end of the array (by adding length).
+ */
+const normalizeIndex = (index: number | undefined, length: number) =>
+	// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+	index === Infinity || index === -Infinity || (index! |= 0) >= 0 ? index! : index! + length;
+
+/* eslint-enable no-bitwise */
+/* eslint-enable no-param-reassign */
+
 export function makeField(
 	context: ProxyContext,
 	fieldSchema: FieldSchema,
 	cursor: ITreeSubscriptionCursor,
 ): EditableField {
-	const targetSequence = new FieldProxyTarget(context, fieldSchema, cursor);
-	return adaptWithProxy(targetSequence, fieldProxyHandler);
+	const target = new FieldProxyTarget(context, fieldSchema, cursor);
+	return adaptWithProxy(target, fieldProxyHandler);
 }
```

Comment:
Sorry about the gratuitous change.  I'll put it back.

Reply:
I made an attempt at extending the Proxy to intercept `slice()`, etc. before it goes through Proxy dispatch.

It kind of works, but breaks things like Object.keys(), which elides the inherited keys from the Proxy.